cassis/cas.py

```diff
@@ -514,8 +573,36 @@ def to_xmi(self, path: Union[str, Path, None] = None, pretty_print: bool = False
         """
         from cassis.xmi import CasXmiSerializer
 
-        serializer = CasXmiSerializer()
+        return self.serialize(CasXmiSerializer(), path, pretty_print)
+
+    def to_json(self, path: Union[str, Path, None] = None, pretty_print: bool = False) -> Optional[str]:
+        """Creates a JSON representation of this CAS.
 
+        Args:
+            path: File path, if `None` is provided the result is returned as a string
+            pretty_print: `True` if the resulting JSON should be pretty-printed, else `False`
+
+
+        Returns:
+            If `path` is None, then the JSON representation of this CAS is returned as a string
+
+        """
+        from cassis.json import CasJsonSerializer
+
+        return self.serialize(CasJsonSerializer(), path, pretty_print)
+
+    def serialize(self, serializer, path: Union[str, Path, None] = None, pretty_print: bool = False):
```

Comment:
Does this need to be public? Also, a type or protocol for serializers would be nice

Reply:
Doesn't need to be public. What is a protocol in this particular context? You mean like an interface?

Reply:
You can then let it start with an underscore

https://docs.python.org/3/library/typing.html#typing.Protocol